homeassistant/components/togrill/sensor.py

```diff
@@ -0,0 +1,139 @@
+
+from __future__ import annotations
+
+from collections.abc import Callable, Mapping
+from dataclasses import dataclass
+from typing import Any, cast
+
+from togrill_bluetooth.packets import Packet, PacketA0Notify, PacketA1Notify
+
+from homeassistant.components.sensor import (
+    SensorDeviceClass,
+    SensorEntity,
+    SensorEntityDescription,
+    SensorStateClass,
+    StateType,
+)
+from homeassistant.const import PERCENTAGE, EntityCategory, UnitOfTemperature
+from homeassistant.core import HomeAssistant
+from homeassistant.helpers.entity_platform import AddConfigEntryEntitiesCallback
+
+from . import ToGrillConfigEntry
+from .const import CONF_PROBE_COUNT
+from .coordinator import ToGrillCoordinator
+from .entity import ToGrillEntity
+
+
+@dataclass(kw_only=True, frozen=True)
+class ToGrillSensorEntityDescription(SensorEntityDescription):
+    """Description of entity."""
+
+    has_entity_name: bool = True
+    packet_type: int
+    packet_extract: Callable[[Packet], StateType]
+    entity_supported: Callable[[Mapping[str, Any]], bool] = lambda _: True
+
+
+def _get_temperature_description(probe_number: int):
+    def _get(packet: Packet) -> StateType:
+        assert isinstance(packet, PacketA1Notify)
+        if len(packet.temperatures) < probe_number:
+            return None
+        temperature = packet.temperatures[probe_number - 1]
+        if temperature is None:
+            return None
+        return temperature
+
+    def _supported(config: Mapping[str, Any]):
+        return probe_number <= config[CONF_PROBE_COUNT]
+
+    return ToGrillSensorEntityDescription(
+        key=f"temperature_{probe_number}",
+        translation_key="temperature",
+        translation_placeholders={"probe_number": f"{probe_number}"},
+        device_class=SensorDeviceClass.TEMPERATURE,
+        native_unit_of_measurement=UnitOfTemperature.CELSIUS,
+        state_class=SensorStateClass.MEASUREMENT,
+        suggested_display_precision=1,
+        packet_type=PacketA1Notify.type,
+        packet_extract=_get,
+        entity_supported=_supported,
+    )
+
+
+def _get_version_string(packet: Packet) -> str:
+    assert isinstance(packet, PacketA0Notify)
+    return f"{packet.version_major}.{packet.version_minor}"
+
+
+ENTITY_DESCRIPTIONS = (
+    ToGrillSensorEntityDescription(
+        key="battery",
+        translation_key="battery",
+        device_class=SensorDeviceClass.BATTERY,
+        native_unit_of_measurement=PERCENTAGE,
+        entity_category=EntityCategory.DIAGNOSTIC,
+        state_class=SensorStateClass.MEASUREMENT,
+        suggested_display_precision=0,
+        packet_type=PacketA0Notify.type,
+        packet_extract=lambda packet: cast(PacketA0Notify, packet).battery,
+    ),
+    ToGrillSensorEntityDescription(
+        key="version",
+        translation_key="version",
+        entity_category=EntityCategory.DIAGNOSTIC,
+        packet_type=PacketA0Notify.type,
+        packet_extract=_get_version_string,
+    ),
```

Comment:
What kind of version is this? This looks like something we should include in the device info instead

Reply:
Hmm. Yes that would make more sense. Will remove that entity.